packages/core/src/messages/responses/cache-sorted-set-get-scores.ts

```diff
@@ -1,67 +1,42 @@
 import {SdkError, UnknownError} from '../../errors';
 import {
   ResponseBase,
-  ResponseHit,
-  ResponseMiss,
-  ResponseError,
+  BaseResponseMiss,
+  BaseResponseError,
 } from './response-base';
-import * as CacheSortedSetGetScoreResponse from './cache-sorted-set-get-score';
 import {
   _ECacheResult,
   _SortedSetGetScoreResponsePart,
 } from './grpc-response-types';
+import {CacheSortedSetGetScoresResponse} from './enums';
+import {CacheSortedSetGetScore} from '../../index';
 
-type CacheSortedSetGetScoreResponseType =
-  | CacheSortedSetGetScoreResponse.Hit
-  | CacheSortedSetGetScoreResponse.Miss
-  | CacheSortedSetGetScoreResponse.Error;
+interface IResponse {
+  value(): Record<string, number> | undefined;
+  readonly type: CacheSortedSetGetScoresResponse;
+}
 
 /**
- * Parent response type for a sorted set get scores request.  The
- * response object is resolved to a type-safe object of one of
- * the following subtypes:
- *
- * - {Hit}
- * - {Miss}
- * - {Error}
- *
- * `instanceof` type guards can be used to operate on the appropriate subtype.
- * @example
- * For example:
- * ```
- * if (response instanceof CacheSortedSetGetScores.Error) {
- *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
- *   // `CacheSortedSetGetScores.Error` in this block, so you will have access to the properties
- *   // of the Error class; e.g. `response.errorCode()`.
- * }
- * ```
+ * Indicates that the requested data was successfully retrieved from the cache.  Provides
+ * `value*` accessors to retrieve the data in the appropriate format.
  */
-export abstract class Response extends ResponseBase {
-  public value(): Record<string, number> | undefined {
-    if (this instanceof Hit) {
-      return (this as Hit).value();
-    }
-    return undefined;
-  }
-}
-
-class _Hit extends Response {
-  public _responses: CacheSortedSetGetScoreResponseType[] = [];
+export class Hit extends ResponseBase implements IResponse {
+  public _responses: CacheSortedSetGetScore.Response[] = [];
```

Comment:
should this be private? Feels weird to have it public when it starts with `_`

Reply:
agreed thanks